meteor/sofie-server/data-access/repositories/mongo/mongo-part-repository.ts

```diff
@@ -33,4 +35,22 @@ export class MongoPartRepository extends BaseMongoRepository implements PartRepo
 			})
 		)
 	}
+
+	public async deleteSegmentParts(segmentId: string): Promise<void> {
+		this.assertDatabaseConnection(this.deleteSegmentParts.name)
+		const parts: Part[] = await this.getParts(segmentId)
+
+		parts.map((part: Part) => part.id).forEach(async (id: string) => this.pieceRepository.deletePartPieces(id))
+		const partsDeletedResult: DeleteResult = await this.getCollection().deleteMany({ segmentId: segmentId })
+
+		// TODO: Figure out how to archive a 'false' acknowledgment, and add test case using that knowledge
+		if (!partsDeletedResult.acknowledged) {
+			throw new DeletionFailedException(`Deletion of parts was not acknowledged, for segmentId: ${segmentId}`)
+		}
+		if (partsDeletedResult.deletedCount === 0) {
+			throw new DeletionFailedException(
+				`Expected to delete one or more parts, but none was deleted, for segmentId: ${segmentId}`
+			)
+		}
```

Comment:
Isn't a segment with 0 parts technically allowed to exist?

Reply:
Same comment as your similar comment for parts.

For this one i am leaning towards a no however.
What would a segment with 0 parts display when taken?

Reply:
I checked it and it is supported, and it's up to the blueprints to decide what they do with those. We usually hide them but there might be valid reasons to show them e.g. time-anchoring (https://github.com/nrkno/sofie-core/pull/945) or adlibs (shown as a standalone minishelf)

Reply:
You can not take a segment on its own. Only Parts can be taken